### PR TITLE
Non-array params collections are not applicable in expanded form in C# 12 and below.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to https://github.com/dotnet/csharplang/issues/8061
Corresponding spec change - https://github.com/dotnet/csharplang/pull/8077

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs [hunk 1]
```diff
@@ -6804,58 +7111,86 @@ static void Main()
...
-                Diagnostic(ErrorCode.WRN_DynamicDispatchToParamsCollectionIndexer, "new Test4()[d3, d4, d4]").WithLocation(26, 13)
+                Diagnostic(ErrorCode.WRN_DynamicDispatchToParamsCollectionIndexer, "new Test4()[d3, d4, d4]").WithLocation(24, 13)
+                };
+
+            CompileAndVerify(
+                comp,
+                expectedOutput: @"Called2Called1Called3Called5Called3Called4Called7Called6Called8Called9Called9").
+            VerifyDiagnostics(expected);
+
+            comp = CreateCompilation(src2, references: [comp1Ref], targetFramework: TargetFramework.StandardAndCSharp, options: TestOptions.ReleaseExe, parseOptions:TestOptions.RegularNext);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs:7147]:
```suggestion
            comp = CreateCompilation(src2, references: [comp1Ref], targetFramework: TargetFramework.StandardAndCSharp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularNext);
```

**Comment by AlekseyTs** [comment]
@RikkiGibson, @dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@RikkiGibson, @dotnet/roslyn-compiler Please review

**Review by RikkiGibson — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Pull request merged**